src/binding.ts

```diff
@@ -43,13 +43,13 @@ export class BindingService {
         const connectionId = binding.connectionId || binding.serverId || '<default>';
         let connectionBindings = bindings.get(connectionId);
         if (!bindings.has(connectionId)) {
-          bindings.set(connectionId, new Map<string, ProjectBinding[]>());
+          bindings.set(connectionId, new Map());
         }
         connectionBindings = bindings.get(connectionId);
         if (!connectionBindings.has(projectKey)) {
           connectionBindings.set(projectKey, []);
         }
-        connectionBindings.get(projectKey).push(binding);
+        connectionBindings.get(projectKey).push({ folder, binding });
```

Comment:
FYI @kirill-knize-sonarsource - I rebased my branch and updated the service to return the full `WorkspaceFolder` object alongside the binding; this way the folder's URI can be attached to the node in the tree (see below).